diamm/views/contribution.py

```diff
@@ -0,0 +1,50 @@
+from rest_framework import views, renderers, permissions
+from django.http import HttpResponseRedirect
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib import messages
+from diamm.renderers.html_renderer import HTMLRenderer
+from diamm.forms.contribution_form import ContributionForm
+
+
+class MakeContribution(views.APIView):
+    template_name = "website/contribution/contribution.jinja2"
+    renderer_classes = (HTMLRenderer, renderers.JSONRenderer)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+        form = ContributionForm(request.POST)
+        object_id = request.data.get('pk')
+        content_type = request.data.get('type')
+        current_url = request.data.get('from')
+
+        if form.is_valid():
+            contribution = form.save(commit=False)
+            contribution.note = request.data.get('note')
+            contribution.object_id = object_id
+
+            try:
+                contribution.content_type = ContentType.objects.get(app_label="diamm_data", model=content_type)
+            except ObjectDoesNotExist:
+                messages.error(request._request, 'Invalid content type')
+                return HttpResponseRedirect(current_url)
+
+            try:
+                contribution.content_type.model_class().objects.all().get(pk=contribution.object_id)
+            except ObjectDoesNotExist:
+                messages.error(request._request, 'Invalid Object')
+                return HttpResponseRedirect(current_url)
+
+            if not request.user.is_authenticated():
```

Comment:
I'm not sure this is needed -- since `permissions.IsAuthenticated` is set, this method will only ever get called by authenticated users...